snow/consensus/snowball/nnary_snowball.go

```diff
@@ -9,7 +9,22 @@ import (
 	"github.com/ava-labs/avalanchego/ids"
 )
 
-var _ NnarySnowball = (*nnarySnowball)(nil)
+var (
+	_ NnarySnow        = (*nnarySnowball)(nil)
+	_ ConsensusFactory = SnowballFactory{}
+)
+
+type SnowballFactory struct{}
+
+func (SnowballFactory) New(params Parameters, choice ids.ID) NnarySnow {
+	sb := newNnarySnowball(params.BetaVirtuous, params.BetaRogue, choice)
+	return &sb
+}
+
+func (SnowballFactory) NewUnary(params Parameters) UnarySnow {
+	sb := newUnarySnowball(params.BetaVirtuous)
+	return &sb
+}
```

Comment:
Should we have a factory file where we implement both `SnowballFactory` and `SnowflakeFactory`? This file seems like a bit of a weird spot (because the factory creates both `Nnary` and `Unary`)

Reply:
Ya I think it'll be cleaner that way, good idea